package testCases;

import static org.testng.Assert.assertTrue;

import org.testng.Assert;
import org.testng.annotations.Test;

import PageObjects.HomePage;
import PageObjects.PracticePage;
import PageObjects.TestExceptionPage;
import testBase.BaseClass;

public class TC004_TestExceptionAdd extends BaseClass {

	@Test(groups = "regression")
	public void Verify_add_favorite_food() {
		logger.info("***********Starting TC004_TestExceptionAdd ************");

		try {
			HomePage hp = new HomePage(driver);
			hp.clickPracticeMenu();
			logger.info("Clicked on Practice Menu");
			PracticePage pp = new PracticePage(driver);
			pp.clickTestExceptionLink();
			TestExceptionPage te = new TestExceptionPage(driver);
			te.clickEditButton();
			te.inputFavouriteFood("burger");
			te.clickSaveButton();
			if (te.IsFoodSavedMsgDisplayed() == true) {
				logger.info("New food added");
				assertTrue(true);
			} else {
				logger.error("Test failed");
				logger.debug("Remove button not foud");
				assertTrue(false);
			}

		} catch (Exception e) {

			logger.debug("Exception: " + e.getMessage());
			System.out.println(e.getMessage());
			Assert.fail();
		}
		logger.info("***********Ending TC004_TestExceptionAdd************");

	}
}
